src/fcntl.rs

```diff
@@ -596,39 +597,125 @@ pub fn fcntl(fd: RawFd, arg: FcntlArg) -> Result<c_int> {
     Errno::result(res)
 }
 
-// TODO: convert to libc_enum
 #[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
 #[non_exhaustive]
 pub enum FlockArg {
     LockShared,
     LockExclusive,
-    Unlock,
     LockSharedNonblock,
     LockExclusiveNonblock,
-    UnlockNonblock,
 }
 
+/// Represents valid types for flock.
+///
+/// # Safety
+/// Types implementing this must be `!Clone`.
 #[cfg(not(any(target_os = "redox", target_os = "solaris")))]
-pub fn flock(fd: RawFd, arg: FlockArg) -> Result<()> {
```

Comment:
I would rather not remove this function suddenly.  Better would be to leave it in but deprecate it for a release or two.

Reply:
One thing I am concerned is that if we keep the old interface, we keep the original FlockArg type, then we keep those 2 FlockArg::Unlock* variants, which means our user can do:

let lock = Flock::lock(file, Flock::Unlock);

Seems like we should define a new type without Unlock* variants just for the Flock

Reply:
We could make that an EINVAL case if we really wanted to prevent that, or we could just make a note of it in the docs while it's in the deprecated transition stage.

Reply:
EINVAL sounds good to me.